### PR TITLE
3D Demo: Fixes

### LINKED ISSUE #4110: Reload issue with the CGAL demo
Steps: Load polygon mesh, shuffle, orient, reload doesn't reload (can be seen easily if you modify the mesh)

### LINKED ISSUE #4127: Flipping boundary edges crashes the demo
cf title.

### PR DESCRIPTION
**Pull request by maxGimeno**
Fixes #4110, fixes #4127 
- Fix the reloading of an oriented soup. 
- Restore highlighting of self-intersections.

### CODE REVIEW

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.0-Ic-106.shtml

File: Polyhedron/demo/Polyhedron/MainWindow.h [hunk 1]
```diff
@@ -360,7 +360,7 @@ protected Q_SLOTS:
     QString cameraString(CGAL::Three::Viewer_interface *v) const;
   //!Hides not available operations and show available operations in all the
   //!menus.
-  void filterOperations();
+  void filterOperations(bool hide);
```
Comment (sloriot) [inline Polyhedron/demo/Polyhedron/MainWindow.h]: @lrineau do you agree with this API change?

Comment (lrineau) [reply Polyhedron/demo/Polyhedron/MainWindow.h]: I generally no longer like Boolean parameters. But this one is not part of any API. That is more like an ugly implementation detail. I cannot say I agree with the change, but I do not object to its merge.

Comment (maxGimeno) [reply Polyhedron/demo/Polyhedron/MainWindow.h]: It IS an ugly workaround for a HELL of a bug, where one implementation only works on Windows and the other doesn't, and I couldn't find any reason why it doesn't work on windows, so I had to resolve to do it like that.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-5.0-Ic-110.shtml

**Pull request merged**